packages/mattermost-redux/src/actions/posts.ts

```diff
@@ -382,6 +383,9 @@ export function deletePost(post: ExtendedPost) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         const state = getState();
         const delPost = {...post};
+        if (post.root_id === '' && isCollapsedThreadsEnabled(state)) {
+            dispatch(decrementThreadCounts(post));
+        }
```

Comment:
This check should be valid for empty strings and if it is not set as well .. it is a bit shorter as well :D

```suggestion

        if (!post.root_id && isCollapsedThreadsEnabled(state)) {
            dispatch(decrementThreadCounts(post));
        }

```